docs/json_schemas/dns_record/v0/provider.json

```diff
@@ -0,0 +1,177 @@
+{
+  "$defs": {
+    "BaseModel": {
+      "properties": {},
+      "title": "BaseModel",
+      "type": "object"
+    },
+    "DnsRecordProvider": {
+      "description": "List statuses for the DNS records informed by the requirer.",
+      "properties": {
+        "dns_domains": {
+          "anyOf": [
+            {
+              "items": {
+                "$ref": "#/$defs/ProviderDomains"
+              },
+              "type": "array"
+            },
+            {
+              "type": "null"
+            }
+          ],
+          "description": "List statuses for the domains requested by the requirer.",
+          "title": "Dns Domains"
```

Comment:
Unless there's some reason preventing us from doing so I think we should always say "DNS" rather than "Dns". Applies to quite a few places in this PR.